### PR TITLE
Dynamic style values

### PR DESCRIPTION
**Pull request by ajkopinga**
This pull request contains a major refactor of the code that builds OpenLayers style objects for features. The OlStyler now recycles OpenLayers style objects and sets style properties dynamically at runtime before the style object is returned to OpenLayers. This enables setting style properties that depend on feature attributes, while avoiding the overhead of creating new OpenLayers style object instances.

For now only these properties can accept ogc:PropertyName for dynamic styling (see the updated Flags of the Benelux example for a demonstration):
* PointSymbolizer Mark Size
* PointSymbolizer Mark Rotation

TextSymbolizer has been refactored to make use of this style object recycling as well.

### CODE REVIEW

File: src/Reader.js [hunk 1]
```diff
@@ -76,6 +76,57 @@ function addPropWithTextContent(node, obj, prop, trimText = false) {
   }
 }
 
+function addFilterExpressionProp(node, obj, prop, skipEmptyNodes = true) {
```
Comment (allartk) [inline src/Reader.js]: What is the purpose of this function? 
Link to: http://schemas.opengis.net/filter/1.1.0/expr.xsd for documentation?

File: src/Utils.js [hunk 2]
```diff
@@ -78,107 +78,40 @@ export function getRules(featureTypeStyle, feature, resolution, options = {}) {
 }
 
 /**
- * Go through all rules with an external graphic matching the image url
- * and update the __loadingState metadata for the symbolizers with the new imageLoadState.
- * This action replaces symbolizers with new symbolizers if they get a new __loadingState.
- * @param {object} featureTypeStyle A feature type style object.
- * @param {string} imageUrl The image url.
- * @param {string} imageLoadState One of 'IMAGE_LOADING', 'IMAGE_LOADED', 'IMAGE_ERROR'.
+ * Function to memoize style conversion functions that convert sld symbolizers to OpenLayers style instances.
```
Comment (allartk) [inline src/Utils.js]: only used in /styles (maybe it should be there somewhere)

File: src/OlStyler.js [hunk 3]
```diff
@@ -1,525 +1,33 @@
...
+ * @param {Array<ol/style>} styles Array of OL styles.
+ * @param {object|Array<object>} symbolizers Feature symbolizer object, or array of feature symbolizers.
+ * @param {ol/feature} feature OpenLayers feature.
+ * @param {Function} styleFunction Function for getting the OL style object. Signature (symbolizer, feature) => OL style.
  */
-function pushTo(array, item, cache) {
-  if (Array.isArray(item)) {
-    for (let k = 0; k < item.length; k += 1) {
-      array.push(cache(item[k]));
+function appendStyle(styles, symbolizers, feature, styleFunction) {
```
Comment (allartk) [inline src/OlStyler.js]: I think: symbolyzers shouldn't be of mixed type (see code from previous pull request)

File: src/Reader.js (hunk 1 above)
Comment (ajkopinga) [reply src/Reader.js]: I've added documentation for addFilterExpressionProp in b127dd4

File: src/Utils.js (hunk 2 above)
Comment (ajkopinga) [reply src/Utils.js]: I've moved style-related utility functions to a separate styleUtils.js module in bde7add.

File: src/OlStyler.js (hunk 3 above)
Comment (ajkopinga) [reply src/OlStyler.js]: I agree, but that's a refectoring worth its own branch.

**Pull request merged**